lib/src/app/app.dart

```diff
@@ -2,20 +2,30 @@ import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
 
-import '../home/cubits/hello_cubit.dart';
+import '../../l10n/tr.dart';
 import '../home/home_page.dart';
+import '../register/cubits/register_cubit.dart';
+import '../register/register_page.dart';
 
 class App extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      title: 'Tazkrtak',
-      localizationsDelegates: AppLocalizations.localizationsDelegates,
-      supportedLocales: AppLocalizations.supportedLocales,
-      home: BlocProvider(
-        create: (_) => HelloCubit()..fetchMessage(),
-        child: HomePage(),
-      ),
+    return BlocProvider<RegisterCubit>(
+      create: (_) => RegisterCubit(),
+      child: MaterialApp(
+          title: 'Tazkrtak',
+          localizationsDelegates: AppLocalizations.localizationsDelegates,
+          supportedLocales: AppLocalizations.supportedLocales,
+          home: BlocBuilder<RegisterCubit, RegisterState>(
+              builder: (context, state) {
+            if (state is RegisterLoading) {
+              return RegisterPage();
+            } else if (state is RegisterSucess) {
+              return HomePage();
+            } else {
+              return Text(tr(context).error_generic);
+            }
```

Comment:
This code will change after adding the new cubit mentioned below.
See the sent template also for better implementation.